Source/WTF/wtf/glib/GSpanExtras.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include <wtf/glib/GSpanExtras.h>
+
+namespace WTF {
+
+GMallocSpan<char*, GStrv> gKeyFileGetKeys(GKeyFile* keyFile, const char* groupName, GUniqueOutPtr<GError>& error)
+{
+    ASSERT(keyFile);
+    ASSERT(groupName);
+
+    size_t keyCount = 0;
+    GStrv keys = g_key_file_get_keys(keyFile, groupName, &keyCount, &error.outPtr());
+    return adoptGMallocSpan<char*, GStrv>(keys, keyCount);
```

Comment:
The GStrv typedef is intended for use with g_autofree. It doesn't add any value in C++; it just obscures the actual type `char**`. I've never much liked typedefs that hide pointers.

So yeah, my style suggestion is:

```suggestion
GMallocSpan<char*, char**> gKeyFileGetKeys(GKeyFile* keyFile, const char* groupName, GUniqueOutPtr<GError>& error)
{
    ASSERT(keyFile);
    ASSERT(groupName);

    size_t keyCount = 0;
    char** keys = g_key_file_get_keys(keyFile, groupName, &keyCount, &error.outPtr());
    return adoptGMallocSpan<char*, char**>(keys, keyCount);
```